test/parallel/test-assert-fail.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// no args
+assert.throws(
+  assert.fail.bind(assert),
+  /^AssertionError: undefined undefined undefined$/
+);
+
+// one arg = message
+assert.throws(
+  assert.fail.bind(assert, 'custom message'),
+  /^AssertionError: custom message$/
+);
+
+// two args only, operator defaults to '!='
+assert.throws(
+  assert.fail.bind(assert, 'first', 'second'),
+  /^AssertionError: 'first' != 'second'$/
+);
+
+// three args
+assert.throws(
+  assert.fail.bind(assert, 'ignored', 'ignored', 'another custom message'),
+  /^AssertionError: another custom message$/
+);
+
+// no third arg (but a fourth arg)
+assert.throws(
+  assert.fail.bind(assert, 'first', 'second', undefined, 'operator'),
```

Comment:
I’d personally prefer arrow functions over `.bind` for all of these… no strong opinion, though :)